tests/IntegrationTests/WebAppUiTests/UiTestHelpers.cs

```diff
@@ -1,70 +1,275 @@
+                }
+            }
+            return Process.Start(processStartInfo);
+        }
+
+        /// <summary>
+        /// Builds the path to the process's directory
+        /// </summary>
+        /// <param name="testAssemblyLocation">The path to the test's directory</param>
+        /// <param name="appLocation">The path to the processes directory</param>
+        /// <returns></returns>
+        private static string GetApplicationWorkingDirectory(string testAssemblyLocation, string appLocation)
+        {
+            string testedAppLocation = Path.Combine(Path.GetDirectoryName(testAssemblyLocation)!);
+            // e.g. microsoft-identity-web\tests\IntegrationTests\WebAppUiTests\bin\Debug\net6.0
+            string[] segments = testedAppLocation.Split(Path.DirectorySeparatorChar);
+            int numberSegments = segments.Length;
+            int startLastSegments = numberSegments - 3;
+            int endFirstSegments = startLastSegments - 2;
+            return Path.Combine(
+                Path.Combine(segments.Take(endFirstSegments).ToArray()),
+                appLocation,
+                Path.Combine(segments.Skip(startLastSegments).ToArray())
+            );
+        }
+
+        /// <summary>
+        /// Kills the processes in the queue and all of their children
+        /// </summary>
+        /// <param name="processQueue">queue of parent processes</param>
+        public static void KillProcessTrees(Queue<Process> processQueue)
+        {
+            Process currentProcess;
+            while (processQueue.Count > 0)
+            {
+                currentProcess = processQueue.Dequeue();
+                if (currentProcess == null) { continue;}
+
+                foreach (Process child in GetChildProcesses(currentProcess))
+                {
+                    processQueue.Enqueue(child);
+                };
+                currentProcess.Kill();
+                currentProcess.Close();
+            }
+        }
+
+        /// <summary>
+        /// Gets the child processes of a process on Windows
+        /// </summary>
+        /// <param name="process">The parent process</param>
+        /// <returns>A list of child processes</returns>
+        /// <exception cref="NotImplementedException">Thrown if running on an OS other than Windows</exception>
+        public static IList<Process> GetChildProcesses(this Process process)
+        {
+            // Validate platform compatibility
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                ManagementObjectSearcher processSearch = new ManagementObjectSearcher($"Select * From Win32_Process Where ParentProcessID={process.Id}");
+                IList<Process> processList = processSearch.Get()
+                    .Cast<ManagementObject>()
+#pragma warning disable CA1416 // This call can not be reached except on Windows due to the enclosing if statement

```

Comment:
would be better to use https://learn.microsoft.com/en-us/dotnet/api/system.runtime.versioning.supportedosplatformattribute?view=net-7.0 if possible here

Reply:
I'll look at this in the next PR